rsc/src/main/scala/rsc/scan/Scanner.scala

```diff
@@ -448,9 +448,13 @@ final class Scanner private (
   }
 
   private def emitIdOrKeyword(): Unit = {
-    val token = keywords.getOrElse(lexeme, ID)
-    val value = if (token == ID) lexeme else null
-    emit(token, value)
+    val lexeme = this.lexeme
+    val token = keywords.get(lexeme)
+    if (token == 0) {
```

Comment:
This feels weird and non-obvious - it looks like null is lifted to 0 here.
I played with this code area because the originala keywords.getOrElseId generates a closure to return a constant value for ID.
I think if we are using the behaviour we should hide it in a method on Keywords

Reply:
This is somewhat obscure, I agree. 

Speaking of the proposed solution, how about instead we just create another constant in Tokens that says `final val NONE = 0` and then use `NONE` here?